scripts/visualize/make_circle_map.R

```diff
@@ -33,6 +33,76 @@ visualize.make_circle_map <- function(viz){
   dev.off()
 }
 
+visualize.make_pie_map <- function(viz){
+  deps <- readDepends(viz)
+  circle_sp <- deps[["sp_circles"]]
+  rad_multiplier <- 8500 # this is projection-specific. 
+  
+  crs <- viz[["visualize_args"]][["crs"]]
+  wu_type <- viz[["visualize_args"]][["wu_type"]]
+  radius_col_name <- paste0("radius_", wu_type)
+  
+  
+  # reorder to plot biggest circles in the back
+  # need na.last as either T or F or they are dropped!!!
+  ordered_i <- sort(circle_sp@data[[radius_col_name]], decreasing = TRUE, 
+                    index.return = TRUE, na.last=FALSE)$ix
+  circle_sp <- circle_sp[ordered_i,]
+  
+  # transform map data
+  circle_sp_transf <- sp::spTransform(circle_sp, sp::CRS(crs))
+  map_data_usa <- sf::st_as_sf(maps::map('usa', fill=TRUE, plot = FALSE))
+  map_data_usa <- sf::st_transform(map_data_usa, crs)
+  map_data_county <- sf::st_as_sf(maps::map('county', fill=TRUE, plot = FALSE))
+  map_data_county <- sf::st_transform(map_data_county, crs)
+  
+  
+  png(viz[["location"]], width = 10, height = 6.0, res=450, units = 'in')
+  
+  par(mai=c(0,0,0,0), omi=c(0,0,0,0), xaxs = 'i', yaxs = 'i')
+  plot(sf::st_geometry(map_data_usa), col = "#f1f1f1", border = "white")
+  plot(sf::st_geometry(map_data_county), col = NA, border = "white", add=TRUE, lwd = 0.5)
+  
+  # start w/ irrigation, fill around the categories (make sure they add up!)
+  # note: they don't add up because we aren't including all
```

Comment:
my guess is we'll want an "Other" category if we decide to include this in the final viz

Reply:
agreed